mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -173,4 +179,25 @@ private void purgeLocalStorageAndExportPom(MavenSession session) {
             }
         });
     }
+
+    private File findFirstDirectoryWithConfigFile(File projectDir, File multiModuleProjectDirectory) {
```

Comment:
actually, it doesn't find the first directory with the config file in all cases - not in the case when the file is not present anywhere - then it returns the root directory. how about: `getFirstDirWithConfigOrRootProjectDir`
or just to explain it in the javadoc

Reply:
If we can have self-explanatory method name it's better than javadoc I believe.